assets/src/edit-story/components/panels/panel/panel.js

```diff
@@ -20,37 +20,93 @@
 import styled from 'styled-components';
 import PropTypes from 'prop-types';
 import { v4 as uuidv4 } from 'uuid';
-import { useState, useCallback } from 'react';
+import { useState, useCallback, useEffect } from 'react';
 
 /**
  * Internal dependencies
  */
 import panelContext from './context';
 
+export const PANEL_COLLAPSED_THRESHOLD = 10;
+
 const Wrapper = styled.section`
   display: flex;
   flex-direction: column;
 `;
 
-function Panel({ initialHeight, name, children }) {
+function Panel({ resizeable, initialHeight, name, children }) {
```

Comment:
I'd really like to see all this new functionality integration tested in a functioning panel with clicks, drags, etc.

Reply:
Filed https://github.com/google/web-stories-wp/issues/840 for this since it might take some time and this PR is blocking two others